torchdata/datapipes/iter/load/fsspec.py

```diff
@@ -49,37 +49,56 @@ class FSSpecFileListerIterDataPipe(IterDataPipe[str]):
 
     def __init__(
         self,
-        root: str,
+        root: Union[str, Sequence[str], IterDataPipe],
         masks: Union[str, List[str]] = "",
     ) -> None:
         _assert_fsspec()
 
-        self.root: str = root
+        if isinstance(root, str):
+            root = [
+                root,
+            ]
+        if not isinstance(root, IterDataPipe):
+            self.datapipe: Union[IterDataPipe, IterableWrapper] = IterableWrapper(root)
```

Comment:
nit: `IterDataPipe` should be sufficient as the type hint.

Reply:
Fixed internally